src/agent/src/config.rs

```diff
@@ -59,6 +67,76 @@ pub struct AgentConfig {
     pub server_addr: String,
     pub unified_cgroup_hierarchy: bool,
     pub tracing: tracer::TraceType,
+    pub blocked_endpoints: HashSet<String>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct AgentConfigBuilder {
```

Comment:
IMO, it should have a sane default for some of the fields, like `hotplug_timeout`, `debug_console_vport`, `log_vport` and `server_addr`.

Reply:
@bergwolf The `AgentConfigBuilder` will only overwrites what's coming from a default `AgentConfig`. In other words, when building an `AgentConfig` from a configuration file, we start from a defaulted `AgentConfig` and the configuration file overwrites the default values. It's like an overlay.